mlflow/gateway/providers/openai.py

```diff
@@ -159,6 +205,52 @@ def _prepare_completion_response_payload(self, resp):
             ),
         )
 
+    async def completions_stream(
+        self, payload: completions.RequestPayload
+    ) -> AsyncIterable[completions.StreamResponsePayload]:
+        from fastapi.encoders import jsonable_encoder
+
+        payload = jsonable_encoder(payload, exclude_none=True)
+        self.check_for_model_field(payload)
+        payload = self._prepare_completion_request_payload(payload)
+
+        stream = send_stream_request(
+            headers=self._request_headers,
+            base_url=self._request_base_url,
+            path="chat/completions",
+            payload=self._add_model_to_payload_if_necessary(payload),
+        )
+
+        async for chunk in handle_incomplete_chunks(stream):
+            chunk = chunk.strip()
+            if not chunk:
+                continue
+
+            data = strip_sse_prefix(chunk.decode("utf-8"))
+            if data == "[DONE]":
+                return
+
+            resp = json.loads(data)
+            yield completions.StreamResponsePayload(
+                id=resp["id"],
+                # The chat models response from OpenAI is of object type "chat.completion.chunk".
+                # Since we're using the completions response format here, we hardcode the
+                # "text_completion_chunk" object type in the response instead
+                object="text_completion_chunk",
+                created=resp["created"],
+                model=resp["model"],
+                choices=[
+                    completions.StreamChoice(
+                        index=idx,
```

Comment:
```suggestion
                        index=c["index"],
```

to preserve the index in the original response.

Reply:
Right, seems like we need another PR to fix the existing endpoints too